### DISCUSSION TITLE
Can I create a "realm admin" that can't see the existence of other realms?

### DISCUSSION THREAD

**Discussion by tomqwpl** [keycloak/keycloak#26378] [Q&A]
I've created a realm, and want to create a user that can administer just that realm. I also want that user not to be able to see the presence of other realms.

I found the process slightly confusing, so I may have done it wrong. 
I created a user in the realm, and assigned it all of the roles from the "realm-management" client. 

With this configuration, logged on as that user, I can see the presence of the other realms. I can apparently then select one of those realms in the "realm selection" drop down in the admin console, though when I attempt to navigate to any of the pages I get the generic "the network response was not OK, refresh to continue" error. Looking in the browser dev tools it's clear it's gotten a 403 response back, which is fair enough. 
I've tried playing with the roles. In particular, there is a tempting role called "query realms". However even if I unassign that I can still apparently list the roles. That role is inherited by a couple of other roles (at least "manage-realm" I think), so then tried making sure I had none of those. So with the "hide inherited roles" tick box unticked (so it's showing inherited roles) it doesn't list that my user has the "query realms" role. However still my test user has a dropdown listing the realms.

I feel I must be doing something wrong. I had kind of assumed that when the documentation talks about each realm having its own admin console, there wouldn't be even the option of navigating to other realms. I certainly wasn't expecting a user to get a dropdown of the other realms.

Is this intended? Is there a way to create a "realm admin" that can't see even the presence of other realms?

**Comment by antikalk** [comment]
It should work just the way you described. But with versions 23.0.0 to 23.0.4 there is a bug that all realms are shown in the dropdown even though the user does not have access to the other once. Have a look at this issue: https://github.com/keycloak/keycloak/issues/25392 

With 23.0.5 or eventually 24.0.0 this issue should be fixed.

**Comment by tomqwpl** [comment]
Ah, OK. That's good to know.
